lib/grape/middleware/base.rb

```diff
@@ -25,11 +25,20 @@ def call(env)
 
       def call!(env)
         @env = env
+        after_response, error = nil
         before
         begin
           @app_response = @app.call(@env)
         ensure
-          after_response = after
+          begin
+            error = catch(:error) do
```

Comment:
you're right, throw is not rescued, so this is not necessary. I removed the extra catch.